### PR TITLE
nixos/test-driver: apply ruff check suggestions

### PR DESCRIPTION
**Pull request by NickCao**
## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by Mic92** [comment]
We should probably also switch to ruff instead of flake.

**Comment by mweinelt** [comment]
Please add `nixosTests.nixos-test-driver` into ruffs passthru tests.

**Comment by NickCao** [comment]
> We should probably also switch to ruff instead of flake8.

We are already on ruff. Maybe get rid of black?

**Comment by NickCao** [comment]
> Please add `nixosTests.nixos-test-driver` into ruffs passthru tests.

It's a list, does it work that way?

**Comment by mweinelt** [comment]
You probably have to inherit from nixosTests.nixos-test-driver then.

**Comment by NickCao** [comment]
Done, also modernized it a bit with the new `build-system` and `dependencies` attributes.

**Review by mweinelt — APPROVED** [review]

File: nixos/lib/test-driver/pyproject.toml [hunk 1]
```diff
@@ -35,11 +35,6 @@ ignore_missing_imports = true
 module = "junit_xml.*"
 ignore_missing_imports = true
 
-[tool.black]
-line-length = 88
-target-version = ['py39']
```
Comment (ambroisie) [inline nixos/lib/test-driver/pyproject.toml:40]: `ruff` defaults to 3.8 until v0.8.0 which defaults to 3.9: should we perhaps be explicit about which version to use?

Comment (NickCao) [reply nixos/lib/test-driver/pyproject.toml:40]: Now targeting python 3.12 (our default version).

**Review by GaetanLepage — APPROVED** [review]
LGTM, sorry for breaking the package :/

**Review by ambroisie — APPROVED** [review]

**Comment by Mic92** [comment]
@ofborg test ferm

**Comment by mweinelt** [comment]
> @ofborg test ferm

Sorry, will not leave master in a broken state until ofborg comes around. The ferm tests works, tested locally.

@NickCao Please squash the fixup commit.

**Comment by NickCao** [comment]
> @NickCao Please squash the fixup commit.

Done

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 358150`

---
### `x86_64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:fast_forward: 1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>

---
### `x86_64-darwin`

---
### `aarch64-darwin`

**Comment by mweinelt** [comment]
@GaetanLepage That was entirely pointless, since it didn't get built in the ruff bump PR either and nothing here changed that.

**Pull request merged**